packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -294,87 +318,115 @@ abstract class ResidentWebRunner extends ResidentRunner {
   }
 
   @override
-  Future<void> debugDumpSemanticsTreeInInverseHitTestOrder() async {
+  Future<bool> debugDumpSemanticsTreeInInverseHitTestOrder() async {
+    if (!supportsServiceProtocol) {
+      return false;
+    }
     try {
       await _vmService
         ?.flutterDebugDumpSemanticsTreeInInverseHitTestOrder(
           isolateId: null,
         );
     } on vmservice.RPCError {
-      return;
+      // do nothing.
     }
+    return true;
   }
 
   @override
-  Future<void> debugToggleDebugPaintSizeEnabled() async {
+  Future<bool> debugToggleDebugPaintSizeEnabled() async {
+    if (!supportsServiceProtocol) {
+      return false;
+    }
     try {
       await _vmService
         ?.flutterToggleDebugPaintSizeEnabled(
           isolateId: null,
         );
     } on vmservice.RPCError {
-      return;
+      // do nothing.
     }
+    return true;
   }
 
   @override
-  Future<void> debugToggleDebugCheckElevationsEnabled() async {
+  Future<bool> debugToggleDebugCheckElevationsEnabled() async {
+    if (!supportsServiceProtocol) {
+      return false;
+    }
     try {
       await _vmService
         ?.flutterToggleDebugCheckElevationsEnabled(
           isolateId: null,
         );
     } on vmservice.RPCError {
-      return;
+      // do nothing.
     }
+    return true;
   }
 
   @override
-  Future<void> debugTogglePerformanceOverlayOverride() async {
+  Future<bool> debugTogglePerformanceOverlayOverride() async {
+    if (!supportsServiceProtocol) {
+      return false;
+    }
     try {
       await _vmService
         ?.flutterTogglePerformanceOverlayOverride(
           isolateId: null,
         );
     } on vmservice.RPCError {
-      return;
+      // do nothing.
     }
+    return true;
   }
 
   @override
-  Future<void> debugToggleWidgetInspector() async {
+  Future<bool> debugToggleWidgetInspector() async {
+    if (!supportsServiceProtocol) {
```

Comment:
Does web need the debug check?

Reply:
The web does not need the debug check, because there is no real "profile" mode with a debug connection but release perf.